lib/Compiler.js

```diff
@@ -2,520 +2,524 @@
 
-				if(!Buffer.isBuffer(content)) {
-					content = new Buffer(content, "utf8"); //eslint-disable-line
+					if(!Buffer.isBuffer(content)) {
+						content = new Buffer(content, "utf8"); // eslint-disable-line
+					}
+
+					source.existsAt = targetPath;
+					source.emitted = true;
+					this.outputFileSystem.writeFile(targetPath, content, callback);
 				}
 
-				source.existsAt = targetPath;
-				source.emitted = true;
-				this.outputFileSystem.writeFile(targetPath, content, callback);
-			}
+			}, err => {
+				if(err) return callback(err);
 
-		}.bind(this), function(err) {
-			if(err) return callback(err);
+				afterEmit.call(this);
+			});
+		}
 
-			afterEmit.call(this);
-		}.bind(this));
-	}
+		function afterEmit() {
+			this.applyPluginsAsyncSeries1("after-emit", compilation, err => {
+				if(err) return callback(err);
 
-	function afterEmit() {
-		this.applyPluginsAsyncSeries1("after-emit", compilation, function(err) {
-			if(err) return callback(err);
+				return callback();
+			});
+		}
 
-			return callback();
-		});
 	}
 
-};
+	emitRecords(callback) {
+		if(!this.recordsOutputPath) return callback();
+		const idx1 = this.recordsOutputPath.lastIndexOf("/");
+		const idx2 = this.recordsOutputPath.lastIndexOf("\\");
+		let recordsOutputPathDirectory = null;
+		if(idx1 > idx2) recordsOutputPathDirectory = this.recordsOutputPath.substr(0, idx1);
+		if(idx1 < idx2) recordsOutputPathDirectory = this.recordsOutputPath.substr(0, idx2);
+		if(!recordsOutputPathDirectory) return writeFile.call(this);
+		this.outputFileSystem.mkdirp(recordsOutputPathDirectory, err => {
+			if(err) return callback(err);
+			writeFile.call(this);
+		});
 
-Compiler.prototype.emitRecords = function emitRecords(callback) {
-	if(!this.recordsOutputPath) return callback();
-	var idx1 = this.recordsOutputPath.lastIndexOf("/");
-	var idx2 = this.recordsOutputPath.lastIndexOf("\\");
-	var recordsOutputPathDirectory = null;
-	if(idx1 > idx2) recordsOutputPathDirectory = this.recordsOutputPath.substr(0, idx1);
-	if(idx1 < idx2) recordsOutputPathDirectory = this.recordsOutputPath.substr(0, idx2);
-	if(!recordsOutputPathDirectory) return writeFile.call(this);
-	this.outputFileSystem.mkdirp(recordsOutputPathDirectory, function(err) {
-		if(err) return callback(err);
-		writeFile.call(this);
-	}.bind(this));
-
-	function writeFile() {
-		this.outputFileSystem.writeFile(this.recordsOutputPath, JSON.stringify(this.records, undefined, 2), callback);
+		function writeFile() {
+			this.outputFileSystem.writeFile(this.recordsOutputPath, JSON.stringify(this.records, undefined, 2), callback);
+		}
 	}
-};
 
-Compiler.prototype.readRecords = function readRecords(callback) {
-	var self = this;
-	if(!self.recordsInputPath) {
-		self.records = {};
-		return callback();
-	}
-	self.inputFileSystem.stat(self.recordsInputPath, function(err) {
-		// It doesn't exist
-		// We can ignore self.
-		if(err) return callback();
+	readRecords(callback) {
+		const self = this;
```

Comment:
`self` was only an alias for `this`. This is no longer needed with arrow functions